boot/boot.js

```diff
@@ -1466,10 +1466,10 @@ $tw.Wiki = function(options) {
 // Dummy methods that will be filled in after boot
 $tw.Wiki.prototype.clearCache =
 $tw.Wiki.prototype.clearGlobalCache =
-$tw.Wiki.prototype.enqueueTiddlerEvent = function() {};
+$tw.Wiki.prototype.enqueueTiddlerEvent = $tw.Wiki.prototype.enqueueTiddlerEvent || function() {};
```

Comment:
Is this change and all the following in this file so these methods can be optionally be defined in the code of the alternate store implementation?

The comment says they "will be filled in after boot". Won't that overwrite the method if it is implemented in the alternate store?

Also with the above it looks like clearCache and clearGlobalCache will also have the same function as enqueueTiddlerEvent if it has an alternate store implementation. Is that intended?

Reply:
> The comment says they "will be filled in after boot". Won't that overwrite the method if it is implemented in the alternate store?

These three methods all have an implementation loaded from wiki.js, but need to be invoked before wiki.js is fully loaded, so we provide this dummy implementation.

> Is this change and all the following in this file so these methods can be optionally be defined in the code of the alternate store implementation?

That's the idea. At the moment the additional methods in wiki.js are loaded unconditionally, and so can't be overridden by alternate stores, which is OK for the moment but might need to be revisited

> Also with the above it looks like clearCache and clearGlobalCache will also have the same function as enqueueTiddlerEvent if it has an alternate store implementation. Is that intended?

These methods are currently overwritten by the implementations in wiki.js, and as above there's currently no support for alternate stores to override those methods.

The wiki store implementation in boot.js has always used a closure for key state information like the hashmap of tiddlers and shadow tiddlers, which makes clear which wiki methods actually have and need access to the raw tiddler store, and to encapsulate the implementation details of the store. That means that it should be possible to just override those methods that have access to the private data in the closure, and seamlessly take advantage of the implementation of the higher level methods in wiki.js.

Reply:
Sounds reasonable and I might still be missing part of your point.

By my thinking these 3 lines should have either remained unchanged since overwriting them would only last until wiki.js is loaded and therefore isn't very useful:

```js
$tw.Wiki.prototype.clearCache =
$tw.Wiki.prototype.clearGlobalCache =
$tw.Wiki.prototype.enqueueTiddlerEvent = function() {};
```

or each should have their own separate overwrite:

```js
$tw.Wiki.prototype.clearCache = $tw.Wiki.prototype.clearCache || function();
$tw.Wiki.prototype.clearGlobalCache = $tw.Wiki.prototype.clearGlobalCache || function();
$tw.Wiki.prototype.enqueueTiddlerEvent = $tw.Wiki.prototype.enqueueTiddlerEvent || function();
```

Reply:
Hi @btheado I think you're right that there's no point in providing an underride for these methods because they will be overwritten later.